Plugins/Events/Events/ResourceEvents.cpp

```diff
@@ -0,0 +1,161 @@
+#include "Events/ResourceEvents.hpp"
+#include "API/CExoBase.hpp"
+#include "API/CExoAliasList.hpp"
+#include "API/CNWSModule.hpp"
+#include "API/CExoResMan.hpp"
+#include "API/Functions.hpp"
+#include "API/Constants.hpp"
+#include "Events.hpp"
+#include "Utils.hpp"
+
+#include <sys/inotify.h>
+#include <unistd.h>
+#include <poll.h>
+#include <climits>
+#include <fcntl.h>
+
+namespace Core {
+    extern bool g_CoreShuttingDown;
+}
+
+namespace Events {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+using namespace NWNXLib::API::Constants;
+
+#define EVENT_SIZE          ( sizeof (struct inotify_event) )
+#define EVENT_BUF_LEN       ( 128 * ( EVENT_SIZE + NAME_MAX + 1) )
+
+static bool processEvents;
```

Comment:
This needs to be marked as `volatile`. Otherwise, the side thread might load the value in a register and keep checking the register instead of loading from memory every time.